File: src/androidTest/java/com/amaze/filemanager/test/DrawerRotationTests.java
package com.amaze.filemanager.test;

import android.support.test.InstrumentationRegistry;
import android.support.test.runner.AndroidJUnit4;
import android.support.test.uiautomator.UiScrollable;
import android.support.test.uiautomator.UiSelector;

import com.amaze.filemanager.R;
import com.amaze.filemanager.test.Utilities.Utils;

import org.junit.After;
import org.junit.Before;
import org.junit.Test;
import org.junit.runner.RunWith;

import static org.junit.Assert.assertTrue;

@RunWith(AndroidJUnit4.class)
public class DrawerRotationTests extends BaseUIAutomatorTest{

    private String quickAccessText, settingsText;

    @Before
    public void setUp(){
        quickAccessText = InstrumentationRegistry.getTargetContext().getString(R.string.quick);
        settingsText = InstrumentationRegistry.getTargetContext().getString(R.string.setting);
    }

    @After
    public void tearDown() throws Exception{
        device.unfreezeRotation();
    }

    @Test
    public void testDrawerOpenRotateScreen() throws Exception{

        Utils.openDrawer(device);

        //assert some elements are visible
        assertElementsDisplayed();

        //rotate screen
        device.setOrientationLeft();

        //assert some elements still visible
        assertElementsDisplayed();

        //rotate screen
        device.setOrientationNatural();

        //assert elements still visible
        assertElementsDisplayed();
    }

    private void assertElementsDisplayed() throws Exception{
        UiScrollable navDrawer = new UiScrollable(
                new UiSelector().resourceId("com.amaze.filemanager:id/menu_drawer"));
        assertTrue("Element " + quickAccessText + " should be visible",
                navDrawer.scrollTextIntoView(quickAccessText));
        assertTrue("Element " + settingsText + " should be visible",
                device.findObject(new UiSelector().text(settingsText)).exists());
    }
}

File: src/tauTests/tests/DrawerRotationTests.py
# -*- coding: utf-8 -*-
from Utilities import TestDataSource, Utils
class DrawerRotationTests(UITestCase):

    def setUp(self):
        global quickAccessText, settingsText
        quickAccessText = "Quick Access"
        settingsText = "Settings"
        
        reactor.add("Permission Reactor", self.allowPermissions, text="Allow")
        launch.activity('com.amaze.filemanager', '.activities.MainActivity', verify=False)
        
    def tearDown(self):
        packages.clearData('com.amaze.filemanager')
        
    def allowPermissions(self):
        tap.resourceId("com.android.packageinstaller:id/permission_allow_button")

    @testCaseInfo('<rotate screen in nav drawer>', deviceCount=1)
    def testDrawerOpenRotateScreen(self):
        
        wait(500)
        #Open drawer
        Utils.openDrawer()
        
        #assert some elements are visible
        self.assertElementsDisplayed()
        
        #rotate screen
        orientation.left()
        
        #assert some elements are visible
        self.assertElementsDisplayed()
        
        #rotate screen
        orientation.portrait()
        
        #assert some elements are visible
        self.assertElementsDisplayed()
        
    def assertElementsDisplayed(self):
        
        navList = get.item.resourceId("com.amaze.filemanager:id/menu_drawer")
        navDrawer = get.item.resourceId("com.amaze.filemanager:id/left_drawer")        
        quickFound = find.text(quickAccessText, area = navList, direction="vertical")
        settingsFound = find.text(settingsText, area = navDrawer, direction="vertical")
        
        #assert quickFound, quickAccessText + " should be visible"
        #assert settingsFound, settingsText + " should be visible"
        verify.text(quickAccessText)
        verify.text(settingsText)
    